src/utils/index.ts

```diff
@@ -51,13 +57,16 @@ export function formatUrl(
 
 export function formatCode(
   code: string,
-  args?: Inputs,
-  optionalParams?: Inputs,
+  paramInputs?: Inputs,
+  optionalParamInputs?: Inputs,
```

Comment:
Same feedback here, it's a bit error-prone to me that the caller of `formatCode` would be expected to merge the optional param defaults with the optional param values provided by the caller. IMO this should happen inside this function so that it can be called directly without requiring a manual "preparation step" before.

In other words, I think this should receive `paramInput`, `optionalParamInputs`, and `optionalParamDefaults`.